MetaMorpheus/EngineLayer/Gptmd/GptmdEngine.cs

```diff
@@ -60,134 +62,124 @@ public static bool ModFits(Modification attemptToLocalize, IBioPolymer protein,
+                                                }
+
+                                                offset += variant.VariantSequence.Length - variant.OriginalSequence.Length;
                                             }
-
-                                            // otherwise back calculate the index to the original protein sequence
-                                            if (modIsBeforeVariant)
+                                            if (!foundSite)
                                             {
-                                                AddIndexedMod(localModDict, pepWithSetMods.Protein.NonVariantProtein.Accession, new Tuple<int, Modification>(indexInProtein - offset, mod));
-                                                System.Threading.Interlocked.Increment(ref localModsAdded);
-                                                foundSite = true;
-                                                break;
+                                                AddIndexedMod(modDict, pepWithSetMods.Protein.NonVariantProtein.Accession, new Tuple<int, Modification>(indexInProtein - offset, mod));
                                             }
-
-                                            offset += variant.VariantSequence.Length - variant.OriginalSequence.Length;
-                                        }
-                                        if (!foundSite)
-                                        {
-                                            AddIndexedMod(localModDict, pepWithSetMods.Protein.NonVariantProtein.Accession, new Tuple<int, Modification>(indexInProtein - offset, mod));
-                                            System.Threading.Interlocked.Increment(ref localModsAdded);
                                         }
                                     }
                                 }
                             }
                         }
                     }
-                }
-                foreach (var kvp in localModDict)
-                {
-                    modDict.AddOrUpdate(kvp.Key, kvp.Value, (key, existingBag) =>
-                    {
-                        foreach (var item in kvp.Value)
-                        {
-                            existingBag.Add(item);
-                        }
-                        return existingBag;
-                    });
-                }
-                System.Threading.Interlocked.Add(ref modsAdded, localModsAdded);
-            });
-
-            return new GptmdResults(this, modDict.ToDictionary(kvp => kvp.Key, kvp => new HashSet<Tuple<int, Modification>>(kvp.Value)), modsAdded);
+                });
+
+                // Convert ConcurrentDictionary to Dictionary with HashSet
+                var finalModDict = modDict.ToDictionary(
+                    kvp => kvp.Key,
+                    kvp => new HashSet<Tuple<int, Modification>>(kvp.Value)
+                );
+                return new GptmdResults(this, finalModDict, modsAdded);
```

Comment:
Do you ever increment mods added?

Reply:
lol. no. but it is not incremented in the master. i just fixed this.